forte/pipeline.py

```diff
@@ -798,6 +812,16 @@ def add(
             # This will ask the job to keep a copy of the gold standard.
             self.evaluator_indices.append(len(self.components))
 
+        if ref_name is None:
+            self._ref_names[ref_name] = []
+            self._ref_names[ref_name].append(len(self.components))
+        elif ref_name in self._ref_names.keys():
```

Comment:
When checking existence of a key in dict, we can simply do `ref_name in self._ref_names`, no need to call the `keys()`